bpaotu/bpaotu/biom.py

```diff
@@ -0,0 +1,121 @@
+from collections import OrderedDict
+import datetime
+import itertools
+
+from .util import val_or_empty
+
+import logging
+logger = logging.getLogger('rainbow')
+
+
+def generate_biom_file(query):
+    rows = []
+    columns = []
+
+    otus = otu_rows(query, rows)
+    samples = sample_columns(query, columns)
+    abundance_table = abundance_tbl(query, rows, columns)
+    shape = (str(len(x)) for x in (rows, columns))
+
+    return (s.encode('utf8')
```

Comment:
the `s.encode()` call here is a detail of the Zip support, it's not generally useful (you're turning `str` to `bytes`) - move the encode call out into the Zip code. otherwise, looks good!

Reply:
Hmm... I think I know what u mean and have tried doing that but get this error:

```
runserver_1     | AttributeError: 'generator' object has no attribute 'encode'
```

I have also tried changing the function to ```zf.writestr()``` but get the same error. 

I.e. I have tried something like the following (from the ```otu_export``` function):

```
zf.writestr('contextual.csv', contextual_csv(query.matching_samples()).encode('utf8'))
zf.write_iter('%s.csv' % (kingdom_label), sample_otu_csv_rows(kingdom_id).encode('utf8'))
```

Have I done something wrong?